projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php

```diff
@@ -18,11 +18,15 @@
 /**
  * Filters the robots.txt contents based on the value of the wpcom_data_sharing_opt_out option.
  *
- * @param string     $output The contents of robots.txt.
- * @param int|string $public The value of the blog_public option.
+ * @param string|null $output The contents of robots.txt.
+ * @param int|string  $public The value of the blog_public option.
  * @return string
  */
-function robots_txt( string $output, $public ): string {
+function robots_txt( ?string $output, $public ): string {
+	if ( ! is_string( $output ) ) {
+		$output = '';
+	}
+
 	$public = (int) $public;
```

Comment:
Instead of this, we may have to do like
```php
$public = (int) get_option( 'blog_public' );
```
WordPress 6.8 started casting the value to bool before calling the hook (see WordPress/wordpress-develop@f01e8109), and wpcom picked that up too (see Automattic/wpcom@d84b7196). If we still want to handle the non-Core "-1" case (treating it as "extra-false" rather than `true` as will be passed) in this hook, we'll need to refetch it.

Reply:
Good catch! Addressed in 49cf5f4187e.